azure/scope/cluster.go

```diff
@@ -254,6 +254,20 @@ func (s *ClusterScope) SubnetSpecs() []azure.SubnetSpec {
 			Role:              s.NodeSubnet().Role,
 		},
 	}
+
+	if s.AzureCluster.Spec.BastionSpec.AzureBastion != nil {
+		azureBastionSubnet := s.AzureCluster.Spec.BastionSpec.AzureBastion.Subnet
```

Comment:
IIUC this means azureBastionSubnet will be reconciled in `subnetsSvc` as well as in `bastionSvc`. Is this intentional?

Reply:
This is actually a very good point! Not sure how the PR ended up managing subnet and public IP that way, but I removed my handlers and let the other services reconcile the subnet and public IP.
It feels much better now, thanks @shysank

Reply:
Can we leave bastion subnets and bastion public ip reconciliation to `subnteSvc` and `publicIpSvc` respectively and have `bastionSvc` only reconcile bastion host? Since we reconcile subnets and publicIps  are reconciled before bastion, you should not have to check for their existence, and it will make the code much simpler as well.  wdyt?

Reply:
@whites11 Just checking to see if you need any help or is this something you feel is not needed? 
cc: @CecileRobertMichon for more 👀

Reply:
makes sense to reconcile bastion subnets and bastion public ip by the subnteSvc and publicIpSvc